hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/analysis/HoodieAnalysis.scala

```diff
@@ -120,68 +136,75 @@ object HoodieAnalysis {
     rules
   }
 
+  private[sql] object MatchInsertIntoStatement {
+    def unapply(plan: LogicalPlan): Option[(LogicalPlan, Map[String, Option[String]], LogicalPlan, Boolean, Boolean)] =
+      sparkAdapter.getCatalystPlanUtils.unapplyInsertIntoStatement(plan)
+  }
+
+  private[sql] object ResolvesToHudiTable {
+    def unapply(plan: LogicalPlan): Option[CatalogTable] =
+      sparkAdapter.resolveHoodieTable(plan)
+  }
+
+  private[sql] def failAnalysis(msg: String): Nothing = {
+    throw new AnalysisException(msg)
+  }
 }
 
 /**
  * Rule for convert the logical plan to command.
  *
  * @param sparkSession
  */
-case class HoodieAnalysis(sparkSession: SparkSession) extends Rule[LogicalPlan]
-  with SparkAdapterSupport {
+case class HoodieAnalysis(sparkSession: SparkSession) extends Rule[LogicalPlan] {
```

Comment:
This is the core of the change:

 - Here we remove implementation of the bespoke resolution rules for Hudi components, instead relying on Spark to resolve these (most of these constructs don't have any custom logic relative to vanilla Spark SQL and therefore would be perfectly fine resolved by standard Spark resolution rules) 
 - Instead we repurpose these rules to serve as conversion point from Spark's standard constructs (like `MergeIntoTable`) into Hudi's ones that implement Hudi-specific semantic (`MergeIntoHoodieTableCommand`). Note that, we require these constructs be fully resolved prior to us trying to convert them